### PR TITLE
Add setMemory function to Javascript Agent

### PR DESCRIPTION
**Pull request by Enfop**
Sorry for the delay this PR addresses #1527. I think there is a better solution but i'm not sure which way to go with this.memory
 
If this.memory is adjusted you have this.memory(object) setting the memory and this.memory(key) getting a value from memory which i find confusing. 

Introducing new functions like setMemory and getMemory would bring some code duplication for backward compability.

### CODE REVIEW

File: app/models/agents/java_script_agent.rb [hunk 1]
```diff
@@ -115,9 +116,12 @@ def execute_js(js_function, incoming_events = [])
       context["doLog"] = lambda { |a, x| log x }
       context["doError"] = lambda { |a, x| error x }
       context["getMemory"] = lambda { |a| memory.to_json }
-      context["setMemory"] = lambda do |a, x, y|
+      context["setKey"] = lambda do |a, x, y|
```
Comment (dsander) [inline app/models/agents/java_script_agent.rb]: Maybe `setMemoryKey`?

File: app/models/agents/java_script_agent.rb [hunk 2]
```diff
@@ -21,6 +21,7 @@ class JavaScriptAgent < Agent
       * `this.memory()`
       * `this.memory(key)`
       * `this.memory(keyToSet, valueToSet)`
+      * `this.setMemory(object)` (sets the memory to the values of the input object)
```
Comment (dsander) [inline app/models/agents/java_script_agent.rb]: Maybe `replaces the Agent's memory with the provided object`?

**Comment by dsander** [comment]
Thanks @Enfop! The changes make sense to me.

**Comment by cantino** [comment]
I think this is absolutely the right direction to be heading.  Thank you for making it clearer!  I think `setMemoryKey` is good.

**Comment by Enfop** [comment]
Renamed the function and changed the setMemory description to @dsander's suggestion which i find more fitting.

**Comment by dsander** [comment]
Nice, thanks @Enfop!

**Pull request merged**